packages/web3-utils/src/utils.js

```diff
@@ -23,10 +23,19 @@
 var _ = require('underscore');
 var BN = require('bn.js');
 var numberToBN = require('number-to-bn');
-var utf8 = require('utf8');
 var Hash = require("eth-lib/lib/hash");
 var ethereumBloomFilters = require('ethereum-bloom-filters');
 
+var utf8Encoder;
+var utf8Decoder;
+if (typeof TextEncoder !== 'undefined' && typeof TextDecoder !== 'undefined') {
+    utf8Encoder = new TextEncoder();
+    utf8Decoder = new TextDecoder();
+} else {
```

Comment:
Just a review note: have added utf8 tests to the headless browser suite and manually verified code passes through this block. (The browser tests don't get captured in the coverage report).